pkg/controller/sync/controller.go

```diff
@@ -718,9 +775,33 @@ func clusterOperations(adapter federatedtypes.FederatedTypeAdapter, selectedClus
 		}
 
 		var operationType util.FederatedOperationType = ""
-		if found {
+
+		// If this is an object for the primary cluster and we're a namespace
+		// kind, then perform a read to get the latest resource version. This
+		// prevents an infinite loop of continually updating finalizers and
+		// then removing finalizers, causing propagatedversion to not keep up
+		// with the template resourceVersions being updated.
+		if isNamespaceKind(kind) {
```

Comment:
Ugh, this is not desirable.  I'm assuming you intend to skip version comparison for namespaces in the host cluster, as we discussed?